evalml/pipelines/pipeline_base.py

```diff
@@ -425,6 +426,60 @@ def feature_importance(self):
         df = pd.DataFrame(importance, columns=["feature", "importance"])
         return df
 
+    def graph_json(self):
+        """Generates a JSON with nodes consisting of the component names and parameters, and edges detailing component relationships.
+
+        Returns:
+            dag_json (str): A serialized JSON representation of a DAG structure.
+        """
+        dag_json = {"Nodes": {"X": "X", "y": "y"}, "Edges": {}}
+
+        for (
+            component_name,
+            component_info,
+        ) in self.component_graph.component_dict.items():
+            dag_json["Nodes"][component_name] = {
+                "class": component_info[0]
+                if isinstance(component_info[0], str)
+                else component_info[0].name,
+                "attributes": self.parameters[component_name]
+                if component_name in self.parameters
+                else None,
```

Comment:
Some classes don't have any parameters that get passed through like `StandardScaler`